src/ray/gcs/gcs_client/accessor.h

```diff
@@ -1009,6 +1009,30 @@ class AutoscalerStateAccessor {
   GcsClient *client_impl_;
 };
 
+/// \class PublisherAccessor
+/// `PublisherAccessor` is a sub-interface of `GcsClient`.
+/// This class includes all the methods that are related to
+/// publishing information to GCS.
+class PublisherAccessor {
```

Comment:
what's the point of these sub-interfaces btw? never quite understood the structure

Reply:
It's just a thin wrapper that does some pre-processing (convert argumetns to protobuf) and post-processing (convert  protobuf to results).